authentication/src/main/kotlin/com/uber/sdk2/auth/internal/AuthProvider.kt

```diff
@@ -44,26 +44,27 @@ class AuthProvider(
 
   override suspend fun authenticate(): AuthResult {
     val ssoConfig = withContext(Dispatchers.IO) { SsoConfigProvider.getSsoConfig(activity) }
-    val parResponse =
-      authContext.prefillInfo?.let {
-        val response =
-          authService.loginParRequest(
-            ssoConfig.clientId,
-            RESPONSE_TYPE,
-            Base64Util.encodePrefillInfoToString(it),
-            ssoConfig.scope ?: "profile",
-          )
-        val body = response.body()
-        body?.takeIf { response.isSuccessful }
-          ?: throw AuthException.ServerError("Bad response ${response.code()}")
-      } ?: PARResponse("", "")
-
-    val queryParams =
-      mapOf(
-        "request_uri" to parResponse.requestUri,
-        "code_challenge" to codeVerifierGenerator.generateCodeChallenge(verifier),
-      )
     return try {
```

Comment:
nit: There's a lot going on here. Consider if any additional abstractions would make sense.

Reply:
Sure. I could work on it separately to break it down into multiple steps